### PR TITLE
Add priority type node for SmartTest 8.

### PR DESCRIPTION
**Pull request by redxeth**
All,

@ThaoHuynhFsl has asked me to submit a change needed by one of his projects.  

It appears to be to add a priority type node to support flows for SMT8.  

I'm personally not familiar with SMT8, so would ask the team to help review this PR to see if it's ok.

Thanks,

Daniel

### CODE REVIEW

**Comment by redxeth** [comment]
bump-- have other stuff from Thao's custom fork that we need in other projects.  So need to get this one merged.

File: lib/origen_testers/smartest_based_tester/v93k_smt8/limits_workbook.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ def add_softbin(number, options = {})
             priority: 2
           }.merge(options)
           attrs = @softbins[number] || {}
+          options[:priority] = options[:priority] || 2
```
Comment (info-rchitect) [inline lib/origen_testers/smartest_based_tester/v93k_smt8/limits_workbook.rb:24]: Is this line necessary?

Comment (priyavadan) [reply lib/origen_testers/smartest_based_tester/v93k_smt8/limits_workbook.rb:24]: '2' kind of redundant here with line 21 also present.

**Review by priyavadan — APPROVED** [review]
just the one redundancy, otherwise looks good to me.

**Comment by priyavadan** [comment]
Thanks @info-rchitect , I missed it. Unit tests and example usage would be nice too!

**Comment by info-rchitect** [comment]
> Thanks @info-rchitect , I missed it. Unit tests and example usage would be nice too!

Yeah I got burned before on `origen_testers` PRs when I didn't add unit tests and examples.  Had to make a patch :/

**Comment by redxeth** [comment]
Ok-- let me take a look at that.   Man doing all work for @ThaoHuynhFsl - he needs to buy me lunch!

**Comment by redxeth** [comment]
Hmm- it's tricky-- ODS is a binary format -- where the Priority column would go.   Trying to write a spec for it but that's not the same.

**Comment by redxeth** [comment]
Wow-- haven't written a spec in a long time-- i forced a failure to test it out (test the tester) and it crashes with a long traceback.  Main errors are : 

    undefined method `extract_expression_lines_at' for RSpec::Core::Formatters::SnippetExtractor:Class (NoMethodError)

    uninitialized constant RSpec::Core::Formatters::SnippetExtractor::NoSuchFileError (NameError)

Anyone see this before?  Anyway I'd like to proceed with adding a new spec check-- but a new spec issue must be added to Origen.

**Comment by pderouen** [comment]
> 
> 
> Wow-- haven't written a spec in a long time-- i forced a failure to test it out (test the tester) and it crashes with a long traceback. Main errors are :
> 
> ```
> undefined method `extract_expression_lines_at' for RSpec::Core::Formatters::SnippetExtractor:Class (NoMethodError)
> 
> uninitialized constant RSpec::Core::Formatters::SnippetExtractor::NoSuchFileError (NameError)
> ```
> 
> Anyone see this before? Anyway I'd like to proceed with adding a new spec check-- but a new spec issue must be added to Origen.

@redxeth that looks like an issue that keeps popping up when a test fails. The fix is to comment out [this](https://github.com/Origen-SDK/origen_testers/blob/65b59dc151e298ca7048baf255d89041c32b853a/spec/spec_helper.rb#L23) line in spec_helper.rb

**Comment by redxeth** [comment]
Thanks!  Should that be checked in ?   Seems we shouldn't have crashing in spec tests.

**Comment by pderouen** [comment]
> 
> 
> Thanks! Should that be checked in ? Seems we shouldn't have crashing in spec tests.

I have been checking it in commented out. If I remember correctly the app generator has that line removed now when you create a new app.

**Review by redxeth — COMMENTED** [review]
> What about unit tests?

Done-- approve ?

**Comment by redxeth** [comment]
Anyway at this point leaving it alone until I better understand the rspec issue better, or someone explains it to me.   Have another PR to follow this one on origentesters.

**Comment by redxeth** [comment]
Ok fixed the crash issue, refer to : https://github.com/Origen-SDK/origen_app_generators/issues/15.

**Comment by redxeth** [comment]
@info-rchitect  let me know if you're ok to approve now.

**Review by info-rchitect — APPROVED** [review]

**Pull request merged**